File: examples/weather-mcp/src/main/java/com/example/weather/model/GetWeatherRequest.java
/*
 * Copyright (c) 2026 Konstantin Pavlov.
 */

package com.example.weather.model;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonTypeName;
import org.jspecify.annotations.Nullable;

@JsonClassDescription("Input for looking up the current weather in a city.")
@JsonTypeName("GetWeatherRequest")
public record GetWeatherRequest(
        @JsonPropertyDescription("City name (e.g., London, Tokyo, New York)") String city,
        @JsonPropertyDescription("Temperature unit (default: celsius)") @Nullable TemperatureUnit units) {
}

File: examples/weather-mcp-kotlin/src/main/kotlin/com/example/weather/model/GetWeatherRequest.kt
/*
 * Copyright (c) 2026 Konstantin Pavlov/IT Staff and contributors.
 */

package com.example.weather.model

import kotlinx.serialization.SerialName
import me.kpavlov.kt.schema.Description

@SerialName("GetWeatherRequest")
data class GetWeatherRequest(
    @Description("City name (e.g., London, Tokyo, New York)")
    val city: String,
    @Description("Temperature unit (default: Celsius)")
    val units: TemperatureUnit = TemperatureUnit.Celsius,
)
